aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -29,58 +29,22 @@ namespace {
 // This means that we allocate a [1,0] size indices tensor and a [0] size
 // values tensor for such an empty tensor.
 SparseTensorImpl::SparseTensorImpl(at::TensorTypeId type_id, at::ScalarType scalar_type)
-    : TensorImpl(type_id, scalar_type, false)
-    , size_{0}
+    : TensorImpl(
+        type_id,
+        scalar_type,
+        /* is_variable */ false,
```

Comment:
is there a reason you didn't include is_variable in TensorImpOptions?  I think you should include it, but make it not a default argument, because unlike the other cases, there's not an obvious default.